File: backend/RockClub.Shared/Dtos/ColaboradorResponseDTO.cs

using System.ComponentModel.DataAnnotations.Schema;


namespace RockClub.Shared.Dtos
{
    public class ColaboradorResponseDTO
    {
        public Guid Colaborador_id { get; set; }
        public string? Nome { get; set; }

        [Column(TypeName = "date")]
        public string Data_nascimento { get; set; }
        public string? Cpf { get; set; }
        public string? Endereco { get; set; }
        public string? Sexo { get; set; }
        public string? Telefone { get; set; }
        public string? Email { get; set; }

        [Column(TypeName = "date")]
        public string Data_admissao { get; set; }
        public string? Cargo { get; set; }
        public decimal? Salario { get; set; }
        public string? Departamento { get; set; }


        public ColaboradorResponseDTO
            (Guid colaborador_id,
            string? nome,
            string data_nascimento,
            string? cpf, 
            string? endereco,
            string? sexo,
            string? telefone,
            string? email,
            string data_admissao,
            string? cargo,
            decimal? salario,
            string? departamento)
        {
            Colaborador_id = colaborador_id;
            Nome = nome;
            Data_nascimento = data_nascimento;
            Cpf = cpf;
            Endereco = endereco;
            Sexo = sexo;
            Telefone = telefone;
            Email = email;
            Data_admissao = data_admissao;
            Cargo = cargo;
            Salario = salario;
            Departamento = departamento;
        }
    }
}

File: frontend/src/app/models/ColaboradorResponseDTO.ts
import { CargoEnum } from "./CargoEnum";
import { DepartamentoEnum } from "./DepartamentoEnum";
import { SexoEnum } from "./SexoEnum";


/*  no banco a resposta vem com as propriedades minusculas então aqui no angular coloque todas as response minusculo para evitar dor de cabeça */

export interface ColaboradorResponseDTO {
  id: string,
  nome: string,
  data_nascimento: string,
  cpf: string,
  endereco: string,
  sexo: SexoEnum,
  telefone: string,
  email: string,
  data_admissao: string,
  cargo: CargoEnum,
  salario: number,
  departamento: DepartamentoEnum
}




